mitiq/vd/vd.py

```diff
@@ -0,0 +1,60 @@
+import cirq
+import numpy as np
+
+def M_copies_of_rho(rho: cirq.Circuit, M: int=2) -> cirq.Circuit:
+    """ Copies a circuit M times in parallel.
+
+    Given a circuit rho that acts on N qubits, this function returns a circuit that copies rho M times in parallel.
+    This means the resulting circuit has N * M qubits.
+
+    Args:
+        rho: 
+            The circuit to be copied.
+        M:
+            The number of copies of rho to be made.
+
+    Returns:
+        A cirq circuit that is the parallel composition of M copies of rho.
+    """
+    
+    # if M <= 1:
+    #     print("warning: M_copies_of_rho is not needed for M <= 1")
+    #     return rho
+
+    N = len(rho.all_qubits())
+
+    circuit = cirq.Circuit()
+    qubits = cirq.LineQubit.range(N*M)
+
+    for i in range(M):
+        circuit += rho.transform_qubits(lambda q: qubits[q.x + N*i])
+
+    return circuit
+
+# add unit tests
+def test_M_copies_of_rho():
```

Comment:
1. This test should be moved to the test file.
2. We should add a separate test to ensure each circuit that comes out of this function has the intended number of qubits.